corehq/motech/repeaters/views/repeat_records.py

```diff
@@ -186,27 +196,44 @@ def _payload_id_and_search_link(self, payload_id):
         )
 
     def _make_row(self, record):
+        checkbox = mark_safe(
+            """<input type="checkbox" onclick="uncheckSelects()" class="xform-checkbox"
+            value="{}" name="xform_ids"/>""".format(record.get_id)
+        )
         row = [
+            checkbox,
             self._make_state_label(record),
             record.repeater.get_url(record) if record.repeater else _('Unable to generate url for record'),
             self._format_date(record.last_checked) if record.last_checked else '---',
             self._format_date(record.next_check) if record.next_check else '---',
             render_to_string('repeaters/partials/attempt_history.html', {'record': record}),
             self._make_view_payload_button(record.get_id),
             self._make_resend_payload_button(record.get_id),
-            self._make_requeue_payload_button(record.get_id) if record.cancelled and not record.succeeded
-            else self._make_cancel_payload_button(record.get_id) if not record.cancelled
-            and not record.succeeded
-            else None
         ]
 
+        if record.cancelled and not record.succeeded:
+            row.append(self._make_requeue_payload_button(record.get_id))
+        elif not record.cancelled and not record.succeeded:
+            row.append(self._make_cancel_payload_button(record.get_id))
+        else:
+            row.append(None)
+
         if toggles.SUPPORT.enabled_for_request(self.request):
-            row.insert(1, self._payload_id_and_search_link(record.payload_id))
+            row.insert(2, self._payload_id_and_search_link(record.payload_id))
         return row
 
     @property
     def headers(self):
         columns = [
+            DataTablesColumn(
+                mark_safe(
+                    """
+                    Select  <a onclick="selectItemsForAllButton()" class="select-visible btn btn-xs btn-default">all</a>
```

Comment:
Same as above, please move the click handler to javascript.

Nitpick: this might as well be a real `<button>` element instead of an `<a>`.

This text should also be translated (probably easier if you move the "Select" text onto the button itself).